src/WhiteLibrary/keywords/application.py

```diff
@@ -26,6 +27,25 @@ def launch_application(self, sut_path, args=None):
         else:
             self.state.app = Application.Launch(sut_path)
 
+    def _attach_application(self, sut_identifier):
+        if self.state._attach_application_timeout == 0:
+            return Application.Attach(sut_identifier)
+
+        hack = {"sut": None}
+
+        def search_application():
```

Comment:
I'd prefer to get this functionality rewritten to be more readable but i couldn't think off any other ways to get around with problem ..